scripts/find-unused-diganostic-messages.mjs

```diff
@@ -0,0 +1,30 @@
+// @ts-check
+// This file requires a modern version of node 14+, and grep to be available.
+
+// node scripts/find-unused-diagnostic-messages.mjs
+import { readFileSync } from "fs";
+import {EOL} from "os";
+import { execSync } from "child_process";
+
+const diags = readFileSync("src/compiler/diagnosticInformationMap.generated.ts", "utf8");
+const startOfDiags = diags.split("export const Diagnostics")[1];
+
+const missingNames = [];
+startOfDiags.split(EOL).forEach(line => {
+    if (!line.includes(":")) return;
+    const diagName = line.split(":")[0].trim();
+
+    try {
+        execSync(`grep -rnw 'src' -e 'Diagnostics.${diagName}'`).toString();

```

Comment:
there's probably a platform-independent package on npm that does recursive file string search, although feature-detecting grep vs findstr would work too.

Reply:
wes thinks ripgrep ( https://www.npmjs.com/package/vscode-ripgrep ) should be a reasonable tradeoff here

Reply:
To avoid the offensive huge cost of this you could just grep all lines that include `Diagnostics.`, print them to a temp file, then search through this file.  Also, even if the names are well behaved, it would be better to add `-F`.  Also, `-n` is redundant AFAICT.

(And doing it with a grep library seems like an overkill too...)